src/System.Private.CoreLib/src/System/Runtime/InteropServices/CustomMarshalers/EnumVariantViewOfEnumerator.cs

```diff
@@ -0,0 +1,102 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections;
+using System.Runtime.InteropServices.ComTypes;
+
+namespace System.Runtime.InteropServices.CustomMarshalers
+{
+    internal class EnumVariantViewOfEnumerator : IEnumVARIANT, ICustomAdapter
+    {
+        public EnumVariantViewOfEnumerator(IEnumerator enumerator)
+        {
+            if (enumerator is null)
+            {
+                throw new ArgumentNullException(nameof(enumerator));
+            }
+
+            Enumerator = enumerator;
+        }
+
+        public IEnumerator Enumerator { get; }
+
+        public IEnumVARIANT Clone()
+        {
+            if (Enumerator is ICloneable clonable)
+            {
+                return new EnumVariantViewOfEnumerator(clonable.Clone() as IEnumerator);
```

Comment:
This is the wrong use of `as`. This will return `null` which will then cause the constructor to throw. The real issue is the type is wrong - so use a C-style case and force the cast exception which will indicate the real issue.